theme/webpack.config.js

```diff
@@ -105,91 +91,84 @@ module.exports = async env => {
                 }
             ]
         },
-        resolve: {
-            modules: [__dirname, 'node_modules'],
-            mainFiles: ['index'],
-            extensions: ['.js']
-        },
+        resolve: await MagentoResolver.configure({
+            paths: {
+                root: __dirname
+            }
+        }),
         plugins: [
-            new WebpackMagentoRootComponentsChunksPlugin(),
+            new MagentoRootComponentsPlugin(),
             new webpack.NoEmitOnErrorsPlugin(),
             new webpack.EnvironmentPlugin({
                 NODE_ENV: isProd ? 'production' : 'development',
-                SERVICE_WORKER_FILE_NAME: magentoEnv.serviceWorkerFileName
+                SERVICE_WORKER_FILE_NAME: 'sw.js'
             })
-        ],
-        devServer: {
-            contentBase: false,
-            https: true,
-            host: magentoEnv.devServerHostname,
-            port: magentoEnv.devServerPort,
-            publicPath: devPublicPath,
-            before(app) {
-                app.use(
-                    proxy(['**', `!${magentoEnv.publicAssetPath}**/*`], {
-                        secure: false,
-                        target: magentoEnv.storeOrigin,
-                        changeOrigin: true,
-                        logLevel: 'debug'
-                    })
-                );
-                staticFileDirs.forEach(dir => {
-                    app.use(
-                        resolve(magentoEnv.publicAssetPath, dir),
-                        express.static(resolve('web', dir))
-                    );
-                });
-                trustCert();
-            },
-            headers: {
-                'Access-Control-Allow-Origin': '*'
-            }
-        },
-        devtool: 'source-map'
+        ]
     };
 
     // modify the default config for production-like environments
     if (isProd) {
-        // disable sourcemaps
-        delete config.devtool;
-
         // add a second entry point for third-party runtime dependencies
         config.entry.vendor = libs;
 
         // add the CommonsChunk plugin to generate more than one bundle
         config.plugins.push(
             new webpack.optimize.CommonsChunkPlugin({
                 names: ['vendor']
+            }),
+            new ServiceWorkerPlugin({
+                env: {
+                    mode: 'production'
+                },
+                paths,
+                serviceWorkerFileName,
+                runtimeCacheAssetPath
             })
         );
 
         // add the UglifyJS plugin to minify the bundle and eliminate dead code
         config.plugins.push(new UglifyPlugin());
-    }
+    } else {
+        config.devServer = await PWADevServer.configure({
+            publicPath,
+            backendDomain,
+            serviceWorkerFileName,
+            paths,
+            id: 'magento-venia'
+        });
+
+        if (process.env.MOCK_GRAPHQL) {
```

Comment:
Can we pull this out into a separate PR? Have some questions/concerns that shouldn't block the rest of this PR from landing

Reply:
@zetlen

Reply:
Yes, please feel free to delete.